streams/src/main/java/org/apache/kafka/streams/state/WindowStore.java

```diff
@@ -27,8 +27,12 @@
 public interface WindowStore<K, V> extends StateStore, ReadOnlyWindowStore<K, V> {
 
     /**
-     * Put a key-value pair with the current record time as the timestamp
-     * into the corresponding window
+     * Use the current record timestamp as the {@code windowStartTimestamp} and
+     * delegate to {@link WindowStore#put(Object, Object, long)}.
+     *
+     * It's highly recommended to use {@link WindowStore#put(Object, Object, long)} instead, as the record timestamp
+     * is unlikely to be the correct windowStartTimestamp in general.
+     *
```

Comment:
I am wondering, if we might want to remove this method? Do use it only at one place and can easily pass the record timestamp there instead? Or is it useful for some special cases? Of course, this would require a KIP and separate PR -- it's might also not be high priority and just creating a JIRA for it might be fine.

WDYT? \cc @guozhangwang @bbejeck

Reply:
Yeah, I think we should remove it. I just didn't bother because I didn't want to worry about the KIP right now.

Right now, the only usages in our codebase are tests, and they could easily be updated.

Reply:
I created https://issues.apache.org/jira/browse/KAFKA-7245

Reply:
Users can create a window store at the PAPI layer and call `put` to it with the current stream time, and it would not necessarily to be the `windowStartTimestamp`, right?

Reply:
No sure what you aim for for the comment? Do you suggest to keep the method?

Reply:
Yeah I was not sure if we should do KAFKA-7245 to remove the API as it would be useful for PAPI users.

Reply:
`put(key,value)` internally calls `put(key,value,context.timestamp())` and this is something PAPI users can do manually -- and I think, it's good to force PAPI users to think about the timestamp to pass in and not just fall back to  `context.timestamp()` without thinking if it's the right thing to do. We can still close the ticket (or discuss) when anybody picks it up -- or do you want to discuss directly and maybe close immediately?